[rocprofiler-compute] ROCm 7.1 release RC1 cherry pick

vedithal-amd opened the pull request:
## Motivation

Add missing commits to rocprofiler-compute 7.1 release branch

- Bugfixes and stability improvements
- Live attach/detach
- Fixes in counters/metric definitions

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

Tests passing locally

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

jbonnell-amd committed:
Replace cmake-format with gersemi in rocprofiler-compute-formatting.yml (#1053)

* Replace cmake-format with gersemi in rocprofiler-compute-formatting.yml

* Run gersemi formatting on CMakeLists.txt files

* Remove .cmake-format.yaml, add .gersemirc file

* Add more options to .gersemirc

* Add new line to .gersemirc

* Add new line to CMakeLists.txt

* Run gersemi again with new options

jamessiddeley-amd committed:
update TCC_EA0_RDREQ_sum def (#1039)

ywang103-amd committed:
change interval for host_trap in unit test to adapt to single kernel (#1064)

abchoudh-amd committed:
Fix test failures (#1059)

* Test fix

* Added path not exists check

vedithal-amd committed:
[rocprofiler-compute] make pmc files deterministic (#1066)

* make pmc files deterministic

cfallows-amd committed:
Refactor roofline binary detection (#933)

* Simplify the roofline binary pickup process by determining which base distribution the system OS is based off of, and select the correct binary.
* Add more OS distribution support to roofline by modifying the detection parameters and adding an AZL binary
* Update changelog to include roofline support additions

---------

Signed-off-by: Carrie Fallows <Carrie.Fallows@amd.com>

xuchen-amd committed:
[rocprofiler-compute][TUI] improve for cross-platform uses (#1007)

systems-assistant[bot] committed:
Live attach/detach and its unit tests (#53)

xuchen-amd committed:
[rocprofiler-compute] Type annotation patch for analysis_db.py (#981)

vedithal-amd committed:
Fix workload path (#1096)

jamessiddeley-amd committed:
fixed function argument dir in apply_filters (#1100)

vedithal-amd committed:
Use ROCM_PATH for sdk library path (#1097)

vedithal-amd committed:
[rocprofiler-compute] Update gfx12 counter definitions (#1003)

* Update gfx12 counter definitions

* Add counter defintions for Navi 4

* Made changes per https://github.com/ROCm/rocm-systems/pull/238 and
  doubled checked register specification

* bugfix

vedithal-amd committed:
Remove rocprofv1/v2 in favour of rocprofiler-sdk (#673)

* Set default rocprof interface as rocprofiler-sdk

* Remove rocrprofv1 and rocprofv2 interfaces

* Remove deprecation notice for rocprof v1/v2/v3 interfaces
  * Make rocprofiler-sdk the default interface and make rocprofv3 interface opt-in using ROCPROF=rocprofv3

* Add deprecation notice for rocprofv3

vedithal-amd committed:
Fix instructions to build standalone binary (#1116)

abchoudh-amd committed:
Bugfixes for test failure (#1106)

- Bugfixes
- Update test instructions using docker

ywang103-amd committed:
attach/detach: change workload of unit test to accommodate SDK's current limitation (#1169)

* add double mode of workload dynamic_share with on remove sleeping and
set ROCP_TOOL_ATTACH=1 for running workload

* add comment in dynamic_shared.hip to exaplain how to use argv

* refactor the attach/detach profiling time in unit tests

vedithal-amd committed:
Fix tests (#1213)

ggottipa-amd committed:
correcting typo. (#1222)

fixes SWDEV-557963.

cfallows-amd reviewed (dismissed):
Verified roofline work is ported.

benrichard-amd committed:
Fix web GUI displaying same section multiple times (#1267)

Some sections were being displayed multiple times in the web GUI.

Code to append the section was nested inside the subsection loop,
so each time a new subsction was appened to the section,
the entire section was appended.

vedithal-amd commented:
This is not needed anymore since 7.1 release branch was updated
